torch/csrc/jit/runtime/static/impl.h

```diff
@@ -151,6 +163,25 @@ class TORCH_API StaticRuntime {
     DCHECK(i < module_->output_regs.size());
     return reg_[module_->output_regs[i]];
   }
+  friend class MemoryPlanner;
+};
+
+class MemoryPlanner {
```

Comment:
can you comment a quick overview about what this does (handles memory allocation at the graph level and provides two methods that are roughly `allocate` and `deallocate`)

Reply:
Will do